src/main/scala/com/cognite/sdk/scala/v1/fdm/common/properties/ReverseDirectRelationConnectionType.scala

```diff
@@ -0,0 +1,27 @@
+package com.cognite.sdk.scala.v1.fdm.common.properties
+
+import io.circe.{Decoder, Encoder}
+
+import java.util.Locale
+
+sealed abstract class ReverseDirectRelationConnectionType extends Product with Serializable
+
+object ReverseDirectRelationConnectionType {
+  case object MultiReverseDirectRelation extends ReverseDirectRelationConnectionType
+  case object SingleReverseDirectRelation extends ReverseDirectRelationConnectionType
+
+  implicit val reverseDirectRelationConnectionTypeDecoder
+      : Decoder[ReverseDirectRelationConnectionType] = Decoder[String].emap {
+    case "multi_reverse_direct_relation" => Right(MultiReverseDirectRelation)
+    case "single_reverse_direct_relation" => Right(SingleReverseDirectRelation)
+    case other => Left(s"Invalid Connection direction: $other")
+  }
+
+  implicit val reverseDirectRelationConnectionTypeEncoder
+      : Encoder[ReverseDirectRelationConnectionType] =
+    Encoder.instance[ReverseDirectRelationConnectionType](p =>
+      io.circe.Json.fromString(
+        p.productPrefix.replaceAll("([a-z])([A-Z]+)", "$1_$2").toLowerCase(Locale.US)
```

Comment:
I know that kind of stuff already exists in this repo, but maybe just do a single match here? Also might be possible to reuse string constant values between encoder and decoder potentially (need to check if it actually works) ~so
```scala
val multiReverseDirectRelationStr = "multi_reverse_direct_relation"
...emap { case `multiReverseDirectRelationStr` => ... }
p match { case MultiReverseDirectRelation => multiReverseDirectRelationStr}
```
maybe there's some even nicer way to do it :sweat_smile: maybe `override def productPrefix` could work so we don't need to regex

Reply:
I think it's nice with the way you suggest, I used something similar.